src/shared_modules/utils/rocksDBWrapper.hpp

```diff
@@ -147,6 +151,85 @@ namespace Utils
             }
         }
 
+        /**
+         * @brief This method tries to repair a RocksDB database and throws an exception if the process fails.
+         *
+         * @param dbPath The database path.
+         */
+        static void repairDB(const std::string& dbPath)
+        {
+            if constexpr (!std::is_same_v<T, rocksdb::DB>)
+            {
+                throw std::runtime_error("RepairDB is only supported for rocksdb::DB");
+            }
+
+            rocksdb::Options options;
+            if (const auto status {rocksdb::RepairDB(dbPath, options)}; !status.ok())
+            {
+                throw std::runtime_error("Failed to repair RocksDB database. Reason: " +
+                                         std::string {status.getState()});
+            }
+        }
+
+        /**
+         * @brief Tries to open a RocksDB database. If the database is corrupted, it tries to repair it. If the process
+         * fails or if the error is of another kind, it throws an exception.
+         *
+         * @param dbPath Database path.
+         * @param repaired Out parameter that indicates if the database was repaired.
+         * @param enableWal Constructor parameter to enable or disable WAL.
+         * @return TRocksDBWrapper<rocksdb::DB> The instance of the RocksDB database in case of success.
+         */
+        TRocksDBWrapper<rocksdb::DB> static openAndRepairBuilder(const std::string& dbPath,
```

Comment:
Could you move this logic to the constructor?

Reply:
I created a different **static** method to give the option to create the wrapper without forcing the repair.
Also, now that the method returns a tuple, I can't make the constructor to return that type.